bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -170,12 +183,14 @@ public String getCurrentPackage() throws AndroidDebugBridgeDeviceException, Inte
 
     public Optional<Boolean> isHDMIOn() throws InterruptedException, AndroidDebugBridgeDeviceException,
             AndroidDebugBridgeDeviceReadException, TimeoutException, ExecutionException {
+        Boolean fallbackHDMI = fallbacks.getOrDefault(HDMI_STATE_CHANNEL, false);
         String result = runAdbShell("cat", "/sys/devices/virtual/switch/hdmi/state");
```

Comment:
You are not avoiding running the command once the fallback is enabled. I'll move the logcat command into a separate private method and call it directly once fallback enabled becomes true.